UnicornRender/video/include/unicorn/video/CameraFpsController.hpp

```diff
@@ -4,38 +4,46 @@
 * (http://opensource.org/licenses/MIT)
 */
 
-#ifndef UNICORN_VIDEO_CAMERA_FPS_CONTROLLER_HPP
-#define UNICORN_VIDEO_CAMERA_FPS_CONTROLLER_HPP
+#ifndef UNICORN_VIDEO_CAMERAFPSCONTROLLER_HPP
+#define UNICORN_VIDEO_CAMERAFPSCONTROLLER_HPP
 
 #include <unicorn/utility/SharedMacros.hpp>
-#include <unicorn/video/Camera.hpp>
+#include <unicorn/video/CameraController.hpp>
 
 namespace unicorn
 {
 namespace video
 {
-/**
- * @brief FPS style camera controller
- */
-class CameraFpsController
+
+/** @brief FPS style camera controller */
+class CameraFpsController : public CameraController
 {
 public:
-    UNICORN_EXPORT CameraFpsController(unicorn::video::Camera& camera);
-    UNICORN_EXPORT void MoveUp(float deltaTime) const;
-    UNICORN_EXPORT void MoveDown(float deltaTime) const;
-    UNICORN_EXPORT void MoveLeft(float deltaTime) const;
-    UNICORN_EXPORT void MoveRight(float deltaTime) const;
-    UNICORN_EXPORT void MoveForward(float deltaTime) const;
-    UNICORN_EXPORT void MoveBackward(float deltaTime) const;
+    UNICORN_EXPORT CameraFpsController(glm::mat4& cameraView);
+    UNICORN_EXPORT void MoveUp(float distance);
+    UNICORN_EXPORT void MoveDown(float distance);
+    UNICORN_EXPORT void MoveLeft(float distance);
+    UNICORN_EXPORT void MoveRight(float distance);
+    UNICORN_EXPORT void MoveForward(float distance);
+    UNICORN_EXPORT void MoveBackward(float distance);
+
+    /** @brief Updates view matrix by taking diff between past and new mouse coordinates */
     UNICORN_EXPORT void UpdateView(double posX, double posY);
-    UNICORN_EXPORT void Scroll(float yoffset) const;
-    float sensitivity, speed;
+
+    /** @brief Sets mouse coordinates without updating view matrix */
+    UNICORN_EXPORT void UpdateViewPositions(double posX, double posY);
+
+    //! Sensitivity of mouse
+    float sensitivity;
 private:
-    double m_lastX, m_lastY, m_yaw, m_pitch;
+    double m_lastX;
+    double m_lastY;
+    double m_yaw;
+    double m_pitch;
     bool m_dirty;
```

Comment:
CameraController already has `m_isDirty`, is this different? If so, please rename both.